engine.py

```diff
@@ -23,200 +19,285 @@
+        elif nuke_version[0] == 6 and nuke_version[1] < 3:
+            self.log_error(msg)
             return
-        elif (nuke_version[0] == 6
-              and nuke_version[1] == 3
-              and nuke_version[2] < 5):
-            self.log_error("Nuke 6.3v5 is the minimum version supported!")
+        elif nuke_version[0] == 6 and nuke_version[1] == 3 and nuke_version[2] < 5:
+            self.log_error(msg)
             return
-        
-        # keep track of if a UI exists
-        self._ui_enabled = nuke.env.get("gui")
 
-        # versions > 9.0 have not yet been tested so show a message to that effect:
+        # Versions > 9.0 have not yet been tested so show a message to that effect.
         if nuke_version[0] > 9 or (nuke_version[0] == 9 and nuke_version[1] > 0):
-            # this is an untested version of Nuke
+            # This is an untested version of Nuke.
             msg = ("The Shotgun Pipeline Toolkit has not yet been fully tested with Nuke %d.%dv%d. "
                    "You can continue to use the Toolkit but you may experience bugs or "
                    "instability.  Please report any issues you see to support@shotgunsoftware.com" 
                    % (nuke_version[0], nuke_version[1], nuke_version[2]))
             
-            # show nuke message if in UI mode, this is the first time the engine has been started
-            # and the warning dialog isn't overriden by the config:
+            # Show nuke message if in UI mode, this is the first time the engine has been started
+            # and the warning dialog isn't overriden by the config.
             if (self._ui_enabled 
                 and not "TANK_NUKE_ENGINE_INIT_NAME" in os.environ
                 and nuke_version[0] >= self.get_setting("compatibility_dialog_min_version", 10)):
                 nuke.message("Warning - Shotgun Pipeline Toolkit!\n\n%s" % msg)
                            
-            # and log the warning
+            # Log the warning.
             self.log_warning(msg)
-            
-        # now prepare tank so that it will be picked up by any new processes
+
+        # Make sure we are not running Nuke PLE!
+        if nuke.env.get("ple"):
+            self.log_error("The Nuke Engine does not work with the Nuke PLE!")
+            return
+
+        # Do our mode-specific initializations.
+        if self.hiero_enabled:
+            self.init_engine_hiero()
+        else:
+            self.init_engine_nuke()
+
+    def init_engine_hiero(self):
+        """
+        The Hiero-specific portion of engine initialization.
+        """
+        self._last_clicked_selection = []
+        self._last_clicked_area = None
+
+    def init_engine_nuke(self):
+        """
+        The Nuke-specific portion of engine initialization.
+        """
+        # Now check that there is a location on disk which corresponds to the context.
+        if self.context.project is None:
```

Comment:
I think we can move this up into `init_engine` - i think it's a valid check for both engines, we just didn't do it previously for hiero for some reason. I guess you could argue that the check is a bit overkill, i don't think this error message fires very often given how toolkit is set up...